components/entitlement/org.wso2.carbon.identity.entitlement.endpoint/src/main/java/org/wso2/carbon/identity/entitlement/endpoint/auth/BasicAuthHandler.java

```diff
@@ -0,0 +1,179 @@
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean isAuthenticated(Message message, ClassResourceInfo classResourceInfo) {
+        // extract authorization header and authenticate.
+        // get the map of protocol headers
+        Map protocolHeaders = (TreeMap) message.get(Message.PROTOCOL_HEADERS);
+        // get the value for Authorization Header
+        List authzHeaders = (ArrayList) protocolHeaders
+                .get(EntitlementEndpointConstants.AUTHORIZATION_HEADER);
+        if (authzHeaders != null) {
+            // get the authorization header value, if provided
+            String authzHeader = (String) authzHeaders.get(0);
+
+            // decode it and extract username and password
+            byte[] decodedAuthHeader = Base64.decode(authzHeader.split(" ")[1]);
+            String authHeader = new String(decodedAuthHeader);
+            String userName = authHeader.split(":")[0];
+            String password = authHeader.split(":")[1];
+            if (userName != null && password != null) {
+                String tenantDomain = MultitenantUtils.getTenantDomain(userName);
+                String tenantLessUserName = MultitenantUtils.getTenantAwareUsername(userName);
+
+                try {
+                    // get super tenant context and get realm service which is an osgi service
+                    RealmService realmService = (RealmService) PrivilegedCarbonContext
+                            .getThreadLocalCarbonContext().getOSGiService(RealmService.class);
+                    if (realmService != null) {
+                        int tenantId = realmService.getTenantManager().getTenantId(tenantDomain);
+                        if (tenantId == -1) {
+                            log.error("Invalid tenant domain " + tenantDomain);
+                            return false;
+                        }
+                        // get tenant's user realm
+                        UserRealm userRealm = realmService.getTenantUserRealm(tenantId);
+                        boolean authenticated = userRealm.getUserStoreManager().authenticate(
+                                tenantLessUserName, password);
+                        if (authenticated) {
+                            // authentication success. set the username for authorization header and
+                            // proceed the REST call
+                            authzHeaders.set(0, userName);
+                            return true;
+                        } else {
+                            if(log.isDebugEnabled()){
+                                log.error("Authentication failed for the user: " + tenantLessUserName
+                                        + "@" + tenantDomain);
+                            }
+                            return false;
+                        }
+                    } else {
+                        if(log.isDebugEnabled()) {
```

Comment:
It is wrong to check if debug enabled and log error. Either just log the error or check if debug enabled and log it as debug log. Please correct this in all the places.